src/lib.rs

```diff
@@ -158,14 +204,13 @@ impl<S: Float + Debug + na::RealField + simba::scalar::RealField> BoundingBox<S>
         self
     }
     /// Add a Point to a Bounding Box, e.g. expand the Bounding Box to contain that point.
-    pub fn insert(&mut self, o: &na::Point3<S>) -> &mut Self {
+    pub fn insert(&mut self, o: &na::Point3<S>) {
```

Comment:
While I agree that there is some inconsistency in how mutations happen (e.g. insert is in-place, but transform is not), I don't see how removing the return value here improves the situation.
I think returning the bbox can be convenient, as it allows chaining mutations.
Furthermore I'm not sure if there is any downside to returning the bbox.
WDYT?

Reply:
If a function can be used both ways, it may be used as a mutation in some part of the user code and as a transformation in another part, which I see as a bad design. So I decided to modify the function to be purely mutable, similar to push method in std::Vec. You cannot use transform method in other way, so its consistent in this way. Also, the insert method makes a "small" change to the structure - it just adds another point to the many others already inside, so it doesn't make sense to return a new bbox and "copy all the points inside" even if that's not really happening. Whereas the transform method completely changes the structure so it seems natural to me that it returns a new instance, but of course that's a matter of opinion.

Reply:
I'm pushing back here, because this will break current users.
While it is fine to break at times, we should only do so, when there is a good justification.
In my opinion this justification could be: Clarifying in-place mutation vs. copy-mutations.

Although even then I think: Both in-place mutation as well as copy mutations could return the bbox to allow chaining. Chaining is just very convenient and I think we should only remove it if we have a strong technical argument.

Reply:
Looking again at my changes from a backwards compatibility perspective, I agree that it would be silly to break code just to change one line of code. Should I revert the commit?

Reply:
Yes - I think reverting 2b6321b4346b08dab1cd6dbd85be9e9ff43c31ba would be good.